manticore/native/cpu/abstractcpu.py

```diff
@@ -376,9 +386,11 @@ def invoke(self, model, prefix_args=None):
 
             msg = "Concretizing due to model invocation"
             if isinstance(src, str):
-                raise ConcretizeRegister(self._cpu, src, msg)
+                raise ConcretizeRegister(self._cpu, src, msg, setstate=e.setstate)
```

Comment:
Tempted to just do a 
`self.PC = self._last_pc`
before raising the new exception. Instead all the setstates changes. It should impact all child states.

Reply:
Fair point, will try that. Thank you for the review!

Reply:
Done!